crates/flowctl/src/draft/author.rs

```diff
@@ -46,13 +47,13 @@ pub async fn upsert_draft_specs(
         expect_pub_id: Option<models::Id>,
     }
 
-    let mut body: Vec<u8> = Vec::new();
-    body.push('[' as u8);
+    // Serialize DraftSpecs directly to JSON without going through
+    // serde_json::Value in order to avoid re-ordering fields which
+    // breaks sops hmac hashes.
+    let mut draft_specs: Vec<String> = vec![];
 
     for row in collections.iter() {
-        if body.len() != 1 {
-            body.push(',' as u8);
-        }
+        let mut body = vec![];
```

Comment:
> nit: can you instead serde_json::to_string directly here?
Sure! I err'd on the side of caution because I wasn't 100% sure whether that would have the same issue as `serde_json::Value` wrt re-ordering fields.

Also swapped back to `.unwrap()` :+1: